app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,41 @@
+class RecipesController < ApplicationController
+  before_action :authenticate_user!, except: %i[public_recipe index]
+  load_and_authorize_resource
+
+  def index
+    @user = User.find(current_user.id)
+    @recipes = @user.recipes
+  end
+
+  def show
+    @recipe = Recipe.find(params[:id])
+    return unless @recipe.public && @recipe.user != current_user
+
+    redirect_to recipes_path, alert: 'You are not authorized to view this recipe.'
+  end
+
+  def public_recipe
+    @recipes = Recipe.where(public: true)
+  end
+
+  def new
+    @recipe = current_user.recipes.build
+    @foods = Food.all
+  end
+
+  def create
+    @recipe = current_user.recipes.build(params.require(:recipe).permit(:name, :description, :public,
+                                                                        :preparation_time, :cooking_time))
+    if @recipe.save
+      redirect_to recipe_path(@recipe.id)
+    else
+      render 'new'
+    end
+  end
```

Comment:
- [ ] NIce adding the changes requested by the previous reviewer however when I try to create a recipe a got the following error please make sure to fix it so that the user can create a recipe without probs 
<img width="1440" alt="Screenshot 2023-05-26 at 18 41 40" src="https://github.com/Fene-87/recipe_app/assets/61790587/7ce9d113-7533-431c-aed8-34f8f142a613">